### DISCUSSION TITLE
Replication crashes with reason changes_reader_died

### DISCUSSION THREAD

**Discussion by simone-marcon** [apache/couchdb#3106] [Q&A]
Created 3 CouchDb node in Azure, each one in a container instance (Linux machine).
When creating a replication doc it crashes after a few minute reporting info changes_reader_died.

I create the replication via curl command:
curl -X POST https://my-couchdb-address/_replicator -H 'Content-Type: application/json' -d '{
	"_id": "my-id",
	"source": "http://my-couchdb-address/database-name",
	"target": "http://my-couchdb-address/database-name",
	"create_target": false,
	"continuous": true
}'

CouchDb installed version is 2.3.1

Log of the error reported:
[couch.log](https://github.com/apache/couchdb/files/5137322/couch.log)

**Comment by wohali** [comment]
Your log does not have enough detail. It looks like you're trying to use a filter incorrectly. For instance, your filter name is `Design%2FfiltroTablet` but that should start with `_design%2F`.

Does the replication work correctly _without_ the filter?

**Reply by simone-marcon** [reply]
the name of the filter correctly starts with _design%2F. It was a typo of the log file.

Also replication without filters does not work correctly. It continuously changes from Running to Retrying state. The error log report the line I already send in attachment.

**Reply by wohali** [reply]
Your logfile only shows attempts to create replication with filters.

Please wipe your logfile, restart CouchDB, and try replication without filters again for a clean logfile.

**Reply by simone-marcon** [reply]
This is my design document. Is the filter part ok?
```
{
   "_id": "_design/zuluDesign",
   "_rev": "46-dd1ca5e5d5c9bfb1c01f36db1e14dcb6",
   "filters": {
       "filtroTablet": "function(doc,req){\n if(doc.idTablet && doc.idTablet == req.query.IdTablet){\n return true; }else{\n return false; }\n}"
   },
   "views": {
       "getMissionByMissionId": {
           "map": "function (doc) {\n  emit(doc.idMissione, doc);\n}"
       },
       "getMissionByPatientId": {
           "map": "function (doc) {\n  emit(doc.idPaziente, doc);\n}"
       },
       "getDocumentById": {
           "map": "function (doc) {\n  emit(doc._id, doc);\n}"
       },
       "getMissionByMissionStatus_TabletId": {
           "map": "function (doc) {\n  emit([doc.statusMissione, doc.idTablet], doc);\n}"
       },
       "getDocumentRevByPatientId": {
           "map": "function (doc) {\n  emit(doc.idPaziente, doc._rev);\n}"
       },
       "getMissionByMissionId_DocumentType": {
           "map": "function (doc) {\n  emit([doc.idMissione, doc.type], doc);\n}"
       },
       "getPagesListByMissionDocumentId_DocumentType": {
           "map": "function (doc) {\n  emit([doc.idMissioneCouch, doc.type], [doc, doc._rev]);\n}"
       },
       "getPageByDocumentId_DocumentType": {
           "map": "function (doc) {\n  emit([doc._id, doc.type], doc);\n}"
       },
       "getPagesListByMissionDocumentId": {
           "map": "function (doc) {\n  emit(doc.idMissioneCouch, doc);\n}"
       }
   },
   "language": "javascript"
}
```

Today I'll try to make some test following your steps.

**Reply by simone-marcon** [reply]
Here are two files containing logs taken in a couchdb instance with a single replication without filters. The first one is with log level setted on ERROR and the second one with DEBUG setted.

Are there any useful additional information?

[couch_error.log](https://github.com/apache/couchdb/files/5160815/couch_error.log)
[couch_debug.log](https://github.com/apache/couchdb/files/5160816/couch_debug.log)

**Reply by wohali** [reply]
```
[error] 2020-09-02T07:20:00.779000Z couchdb@localhost <0.721.0> -------- couch_replicator_auth_session : Could not parse cookie from response headers cookie_format_invalid
[error] 2020-09-02T07:20:10.382000Z couchdb@localhost <0.1947.0> -------- Replicator, request GET to "https://my-couchdb-server-address/zuluedispatch/_changes?feed=continuous&style=all_docs&since=%2274-g1AAAASTeJzF00sKwjAQBuDgsyKCN9ALWNLG0riyN9G8SpXaSjUI3ehN9CZ6E71JTZOC-7rIZkIC-fmYTFIAgJN0OViyXLKE06iUqRSLZgeRexGns5BFfhQuKWUhWJ6dyS4ThbvLU3W7QwCdVVW1T7qkf1AHw9jHkAreMvM_DJ2rSteNZ6A9oYepT5gdT1R7No3H0R4cCj9YWfJsa8-18Yy0hyPEvGVoxZP1VAU3tSjS_fdmEPGQWeqRMT2M6VmbxtrkMYxj385cG9PLmN61aaJNlAVBEEOLpo8x6Xmamv-PIFYJLaP3XyGDmkA%22&timeout=10000" failed. The received HTTP error code is 504
[error] 2020-09-02T07:20:10.726000Z couchdb@localhost <0.2256.0> -------- Replicator, request GET to "https://my-couchdb-server-address/zuluedispatch/03f3d3dc2cb1909697e47f64c200d0b7?revs=true&open_revs=%5B%221-ce2f803a710ccb3f0c65bce1fd1f064a%22%5D&latest=true" failed. The received HTTP error code is 504
```

Do you have a proxy in between the two CouchDBs? CouchDB itself shouldn't ever return a 504, and the cookie parse error suggests there is some sort of mangling of headers going on.

504 is typically a "gateway timeout" error from a proxy.

**Reply by simone-marcon** [reply]
I am using cloudflare and I read another issue (#2369) that make me think of turning off the cloudflare proxy. 

I'll check if the problem solved.

**Reply by simone-marcon** [reply]
Even disabling Cloudflare proxy, the error persist. The reported logs are the same as the previous ones
